src/charm.py

```diff
@@ -92,6 +111,10 @@ def _on_alertmanager_change(self, event):
     def _loki_push_api_alert_rules_changed(self, event):
         self._configure(event)
 
+    def _on_ingress_changed(self, event):
+        self._configure(event)
+        self.loki_provider.update_endpoint()
```

Comment:
If `self.loki_provider.update_endpoint()` is called after `_configure`, then perhaps calling `_configure` is not needed at all?